api_app/script_analyzers/file_analyzers/xlm_macro_deobfuscator.py

```diff
@@ -0,0 +1,56 @@
+import logging
+from XLMMacroDeobfuscator.deobfuscator import process_file
+from api_app.script_analyzers.classes import FileAnalyzer
+from celery.exceptions import SoftTimeLimitExceeded
+
+logger = logging.getLogger(__name__)
+
+
+class XlmMacroDeobfuscator(FileAnalyzer):
+    def set_config(self, additional_config_params):
+        self.passwords_to_check = [""]
+        additional_passwords_to_check = additional_config_params.get(
+            "passwords_to_check", []
+        )
+        if isinstance(additional_passwords_to_check, list):
+            self.passwords_to_check.extend(additional_passwords_to_check)
+        elif isinstance(additional_passwords_to_check, str):
+            self.passwords_to_check.append(additional_passwords_to_check)
+
+    def run(self):
+        results = {}
+        try:
+            for password in self.passwords_to_check:
+                results = self.decrypt(password)
+                if results:
+                    break
+            if not results:
+                results["error"] = "Can't decrypt with current passwords"
+        except SoftTimeLimitExceeded as e:
+            error_message = (
```

Comment:
I missed this before. But if it's not too much trouble, just one more change.
Line 30-36 can be replaced with `self._handle_base_exception("Soft Time Limit Exceeded error")`

Reply:
Awesome. Thanks!